tests/bsim/bluetooth/host/privacy/peripheral/src/dut_rpa_expired.c

```diff
@@ -0,0 +1,178 @@
+	int err;
+	struct bt_le_ext_adv_info info;
+
+	err = bt_le_ext_adv_get_info(adv, &info);
+	if (err) {
+		return false;
+	}
+	printk("%s advertiser[%d] RPA %s\n", __func__, info.id, bt_addr_le_str(info.addr));
+
+	return true;
+}
+
+bool rpa_expired_cb_returns_false(struct bt_le_ext_adv *adv)
+{
+	/* Return false not to rotate the current RPA */
+	int err;
+	struct bt_le_ext_adv_info info;
+
+	err = bt_le_ext_adv_get_info(adv, &info);
+	if (err) {
+		return false;
+	}
+	printk("%s advertiser[%d] RPA %s\n", __func__, info.id, bt_addr_le_str(info.addr));
+
+	return false;
+}
+
+static void create_adv(struct bt_le_ext_adv **adv, int id, bool expired_return)
+{
+	int err;
+	struct bt_le_adv_param params;
+	static struct bt_le_ext_adv_cb cb_adv[] = {
+		{.rpa_expired = rpa_expired_cb_returns_true},
+		{.rpa_expired = rpa_expired_cb_returns_false}
+	};
+
+	memset(&params, 0, sizeof(struct bt_le_adv_param));
+
+	params.options |= BT_LE_ADV_OPT_EXT_ADV;
+	params.id = id;
+	params.sid = 0;
+	params.interval_min = BT_GAP_ADV_FAST_INT_MIN_1;
+	params.interval_max = BT_GAP_ADV_FAST_INT_MAX_1;
+
+	err = bt_le_ext_adv_create(&params, expired_return ? &cb_adv[0] : &cb_adv[1], adv);
+	if (err) {
+		FAIL("Failed to create advertiser (%d)\n", err);
+	}
+}
+
+void start_rpa_advertising(void)
+{
+	int err;
+	size_t bt_id_count;
+
+	/* Enable bluetooth */
+	err = bt_enable(NULL);
+	if (err) {
+		FAIL("Failed to enable bluetooth (err %d\n)", err);
+	}
+
+	err = settings_load();
+	if (err) {
+		FAIL("Failed to enable settings (err %d\n)", err);
+	}
+
+	bt_id_get(NULL, &bt_id_count);
+
+	if (bt_id_count == 1) {
+		int id_a;
+		int id_b;
+
+		printk("No extra identity found in settings, creating new ones...\n");
+
+		id_a = bt_id_create(NULL, NULL);
+		if (id_a != ID_1) {
+			FAIL("bt_id_create id_a failed (err %d)\n", id_a);
+		}
+
+		id_b = bt_id_create(NULL, NULL);
+		if (id_b != ID_2) {
+			FAIL("bt_id_create id_b failed (err %d)\n", id_b);
+		}
+	} else {
+		printk("Extra identities loaded from settings\n");
+	}
+
+	bt_id_get(NULL, &bt_id_count);
+	if (bt_id_count != CONFIG_BT_ID_MAX) {
+		FAIL("bt_id_get returned incorrect number of identities %u\n", bt_id_count);
+	}
+
+	for (int i = 0; i < CONFIG_BT_EXT_ADV_MAX_ADV_SET; i++) {
+		/* Create first 2 advertising sets with one id and for both sets,rpa_expied_cb
+		 * returns true.
+		 * Create remaining 2 sets with different id and last adv set's rpa_expired cb
+		 * returns false.
+		 *
+		 * So for first two adv sets with same id new rpa's will be generated every
+		 * rotation and for last two adv sets with same id, rpa will continue with
+		 * only one rpa through out the rotations since one of the adv set returned
+		 * false.
+		 */
+		switch (i)  {
+		case ADV_SET_INDEX_1:
+		case ADV_SET_INDEX_2:
+			create_adv(&adv_set[i], ID_1, true);
+			break;
+		case ADV_SET_INDEX_3:
+			create_adv(&adv_set[i], ID_2, true);
+			break;
+		case ADV_SET_INDEX_4:
+			create_adv(&adv_set[i], ID_2, false);
```

Comment:
Could you post a follow-up PR where you add another id w/ two other adv sets, that return `false` for the first one and `true` for the second one? Ie started in reverse order so the iterator hits the `false` first.

Reply:
clarify: this comment is not a blocker.

Reply:
Noted. Will create a pull request for the same.

Reply:
https://github.com/zephyrproject-rtos/zephyr/issues/69969